### PR TITLE
Avoid logmanager crash due to /persist/IMGx/log disappearing

### PR DESCRIPTION
**Pull request by eriknordmark**
@srinibas-zededa please take a look.
I can't tell from all the logs whether this is the only issue, but I see it in a few log files.

@rvs Second commit means nim will not wait for 2 minutes if the model is unknown

### CODE REVIEW

**Comment by srinibas-zededa** [comment]
> @srinibas-zededa please take a look.
> I can't tell from all the logs whether this is the only issue, but I see it in a few log files.
> 
> @rvs Second commit means nim will not wait for 2 minutes if the model is unknown

LGTM.

**Comment by rvs** [comment]
Yup @srinibas-zededa -- that's the desired behaviour for nim

File: pkg/pillar/cmd/baseosmgr/handlebaseos.go [hunk 1]
```diff
@@ -316,6 +318,22 @@ func doBaseOsActivate(ctx *baseOsMgrContext, uuidStr string,
...
 }
 
+func removeContent(dirName string) error {
+	locations, err := ioutil.ReadDir(dirName)
+	if err != nil {
+		return err
+	}
+
+	for _, location := range locations {
+		filelocation := dirName + "/" + location.Name()
```
Comment (kalyan-nidumolu) [inline pkg/pillar/cmd/baseosmgr/handlebaseos.go]: Does it make sense to use filepath.Join instead??

File: pkg/pillar/cmd/baseosmgr/handlebaseos.go [hunk 2]
```diff
@@ -316,6 +318,22 @@ func doBaseOsActivate(ctx *baseOsMgrContext, uuidStr string,
 	return changed
 }
 
+func removeContent(dirName string) error {
+	locations, err := ioutil.ReadDir(dirName)
+	if err != nil {
+		return err
```
Comment (kalyan-nidumolu) [inline pkg/pillar/cmd/baseosmgr/handlebaseos.go]: May be add an Errorf statement here?

File: pkg/pillar/cmd/baseosmgr/handlebaseos.go [hunk 3]
```diff
@@ -316,6 +318,22 @@ func doBaseOsActivate(ctx *baseOsMgrContext, uuidStr string,
...
+	}
+
+	for _, location := range locations {
+		filelocation := dirName + "/" + location.Name()
+		err := os.RemoveAll(filelocation)
+		if err != nil {
+			return err
```
Comment (kalyan-nidumolu) [inline pkg/pillar/cmd/baseosmgr/handlebaseos.go]: Maybe add an Errorf statement here?

**Pull request merged**